src/domain/checkForAndPlaceOrder.ts

```diff
@@ -579,7 +556,8 @@ async function _pollLegacy(
 
     const [{ success, returnData }] = lowLevelCall;
 
-    // If the call failed, we throw an error and wrap the returnData as it is done by erigon / geth 🦦
+    // If the call failed, there may be a custom error to provide hints. We wrap the error in a LowLevelError
+    // so that it can be handled in the catch.
     if (!success) {
       throw new LowLevelError("low-level call failed", returnData);
```

Comment:
Both the error and the message seems to generic. It looks very coupled to `multical aggregate3` call, also the message should probably mention it failed the `getTradeableOrderWithSignature`. Lastly, it might be a good idea to print the Tenderly simulation as a ERROR instead of DEBUG in case of a failure. It might make debugging easier (since `DEBUG` level will likely be disabled in production)

Reply:
The `Multicall3.aggregate3Value`, literally conducts a low-level solidity calls (hence the `LowLevelError` name). 

In solidity, you can do a `.call` and this allows the EVM to provide success (or failure, ie. revert), and the return data from the EVM call. It's most definitely very coupled, but contextually, there's only one place that this happens.